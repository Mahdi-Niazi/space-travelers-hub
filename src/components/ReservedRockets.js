import React from 'react';
import PropTypes from 'prop-types';
import { ListGroup } from 'react-bootstrap';

const ReservedRockets = ({ reservedRockets }) => (
  <div className="container">
    {reservedRockets.length === 0 ? (
      <p style={{ color: 'red' }}>No Reserved Rockets</p>
    ) : (
      <ListGroup className="shadow p-3 mb-5 bg-body-tertiary rounded">
        {reservedRockets.map((rocket) => (
          <ListGroup.Item key={rocket.rocket_id}>
            {rocket.rocket_name}
          </ListGroup.Item>
        ))}
      </ListGroup>
    )}
  </div>
);

ReservedRockets.propTypes = {
  reservedRockets: PropTypes.arrayOf(
    PropTypes.shape({
      rocket_id: PropTypes.string.isRequired,
      rocket_name: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default ReservedRockets;
