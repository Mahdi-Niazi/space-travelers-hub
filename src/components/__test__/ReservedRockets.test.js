import React from 'react';
import { render } from '@testing-library/react';
import ReservedRockets from '../ReservedRockets';

describe('ReservedRockets Component', () => {
  it('should match the snapshot with no reserved rockets', () => {
    const reservedRockets = [];

    const { container } = render(
      <ReservedRockets reservedRockets={reservedRockets} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot with reserved rockets', () => {
    const reservedRockets = [
      { rocket_id: '1', rocket_name: 'Rocket 1' },
      { rocket_id: '2', rocket_name: 'Rocket 2' },
      // Add more rockets if needed
    ];

    const { container } = render(
      <ReservedRockets reservedRockets={reservedRockets} />,
    );

    expect(container).toMatchSnapshot();
  });
});
