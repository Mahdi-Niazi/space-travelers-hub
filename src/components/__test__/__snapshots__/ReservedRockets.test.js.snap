// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservedRockets Component should match the snapshot with no reserved rockets 1`] = `
<div>
  <div
    class="container"
  >
    <p
      style="color: red;"
    >
      No Reserved Rockets
    </p>
  </div>
</div>
`;

exports[`ReservedRockets Component should match the snapshot with reserved rockets 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="shadow p-3 mb-5 bg-body-tertiary rounded list-group"
    >
      <div
        class="list-group-item"
      >
        Rocket 1
      </div>
      <div
        class="list-group-item"
      >
        Rocket 2
      </div>
    </div>
  </div>
</div>
`;
